modules/hivemq/src/main/java/org/testcontainers/hivemq/HiveMQContainer.java

```diff
@@ -0,0 +1,551 @@
+package org.testcontainers.hivemq;
+
+import org.apache.commons.io.FileUtils;
+import org.jetbrains.annotations.NotNull;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.slf4j.event.Level;
+import org.testcontainers.containers.ContainerLaunchException;
+import org.testcontainers.containers.GenericContainer;
+import org.testcontainers.containers.wait.strategy.MultiLogMessageWaitStrategy;
+import org.testcontainers.images.builder.ImageFromDockerfile;
+import org.testcontainers.utility.DockerImageName;
+import org.testcontainers.utility.MountableFile;
+
+import java.io.File;
+import java.io.IOException;
+import java.nio.charset.StandardCharsets;
+import java.time.Duration;
+import java.util.concurrent.ConcurrentHashMap;
+import java.util.concurrent.CountDownLatch;
+import java.util.concurrent.TimeUnit;
+import java.util.concurrent.TimeoutException;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+public class HiveMQContainer extends GenericContainer<HiveMQContainer> {
+
+    private static final Logger LOGGER = LoggerFactory.getLogger(HiveMQContainer.class);
+
+    public static final String DEFAULT_HIVEMQ_EE_TAG = "4.7.2";
+    public static final String DEFAULT_HIVEMQ_CE_TAG = "2021.3";
+    public static final DockerImageName DEFAULT_HIVEMQ_EE_IMAGE_NAME = DockerImageName.parse("hivemq/hivemq4").withTag(DEFAULT_HIVEMQ_EE_TAG);
+    public static final DockerImageName DEFAULT_HIVEMQ_CE_IMAGE_NAME = DockerImageName.parse("hivemq/hivemq-ce").withTag(DEFAULT_HIVEMQ_CE_TAG);
+
+    public static final int DEBUGGING_PORT = 9000;
+    public static final int MQTT_PORT = 1883;
+    public static final int CONTROL_CENTER_PORT = 8080;
+    @SuppressWarnings("OctalInteger")
+    private static final int MODE = 0777;
+    private static final @NotNull Pattern EXTENSION_ID_PATTERN = Pattern.compile("<id>(.+?)</id>");
+
+    private final @NotNull ConcurrentHashMap<String, CountDownLatch> containerOutputLatches = new ConcurrentHashMap<>();
+    private volatile boolean controlCenterEnabled = false;
+    private volatile boolean debugging = false;
```

Comment:
why volatile? It is not expected that containers will be configured from multiple threads concurrently, and other fields expect external synchronization

```suggestion
    private boolean controlCenterEnabled = false;
    private boolean debugging = false;
```